pkg/server/datastore/sqlstore/sqlstore.go

```diff
@@ -0,0 +1,133 @@
+package sqlstore
+
+import (
+	"errors"
+	"fmt"
+
+	"gorm.io/gorm"
+)
+
+const (
+	// MySQL database type
+	MySQL = "mysql"
+	// PostgreSQL database type
+	PostgreSQL = "postgres"
+	// SQLite database type
+	SQLite = "sqlite3"
+)
+
+func OpenDB(connectionString, dbtype string) (*gorm.DB, error) {
+	var dialectvar dialect
+
+	switch dbtype {
+	case SQLite:
+		dialectvar = sqliteDB{}
+	case PostgreSQL:
+		dialectvar = postgresDB{}
+	case MySQL:
+		dialectvar = mysqlDB{}
+	default:
+		return nil, fmt.Errorf("unsupported database_type: %s", dbtype)
+	}
+	db, err := dialectvar.connect(connectionString)
+	if err != nil {
+		return nil, errors.New("Error connecting to: %s" + connectionString)
+	}
+	return db, nil
+}
+
+func CreateOrganizationTableInDB(db *gorm.DB) error {
+	// Creates the Model for the Bridge
+	result := db.AutoMigrate(&Organization{})
+	if result != nil {
+		return fmt.Errorf("sqlstore error: %v", result)
+	}
+	return nil
+}
+
+func InsertOrg(db *gorm.DB, org Organization) error {
+	// Insert new ORG into the DB
+	if err := db.Where(&org).FirstOrCreate(&org).Error; err != nil {
+		return fmt.Errorf("sqlstorage error: %v", err)
+	}
+	return nil
+}
+
+func RetrieveOrg(db *gorm.DB, org *Organization) error {
+	// Insert new ORG into the DB
```

Comment:
```suggestion
// RetrieveOrg retrieves an Organization from the Database. returns an error if something goes wrong.
func RetrieveOrg(db *gorm.DB, org *Organization) error {
```
Having this comment for the other functions will help as well. I advise you to put comments in the other functions. It provides a better description of the function even if it is called outside the file.

Reply:
ok